src/plot.cpp

```diff
@@ -315,6 +315,11 @@ void Plot::set_output_path(pugi::xml_node plot_node)
   } else {
     filename = fmt::format("plot_{}", id());
   }
+  const std::string dir_if_present =
+    filename.substr(0, filename.find_last_of("/") + 1);
+  if (!dir_exists(dir_if_present) and dir_if_present.size() > 0) {
```

Comment:
Performing the simple check first for short-circuit of the logical expression.

```suggestion
  if (dir_if_present.size() > 0 && !dir_exists(dir_if_present)) {
```